index.ts

```diff
@@ -47,21 +49,44 @@ export function isCallstackEvent(event: CallstackEvent): event is CallstackEvent
 export class HawkTracerClient {
     private _client: any;
     private _klass_names: (string | undefined)[] = [];
+    private _timeout?: NodeJS.Timeout;
+    private _max_retries: number;
 
-    constructor(option: HawkTracerClientOptions | string) {
+    constructor(option: HawkTracerClientOptions | string, max_retries: number = 100) {
```

Comment:
Why not add it to HawkTracerClientOptions?

Reply:
It shouldn't be passed as HawkTracerClientOptions, because it doesn't get passed to the native client. It's a separate responsibility, specific to the binding. 

If we were to ram it into HawkTracerClientOptions, then these have to be deconstructed. May give people impression that it's passed down to the C++ code.